### PR TITLE
Activity for displaying all available games

### LINKED ISSUE #5: List of the available games
Global list of games
You can click on a game and see his description
Under the description there is a button "Register"

- [ ] Requires game database

### PR DESCRIPTION
**Pull request by rykeuse**
Closes #5 
- Show list of available games
- class Game is a dummy class (real one implemented by @DonnA98 : #68)

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Override
+    public void onBindViewHolder(@NonNull GameHolder gameHolder, int i) {
+        gameHolder.setDetails(games.get(i));
+    }
+
+    @Override
+    public int getItemCount() {
+        return games.size();
+//        return 15;
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/model/GameAdapter.java]: Remove this line, please

File: app/src/main/res/layout/activity_game_list.xml [hunk 2]
```diff
@@ -1,9 +1,19 @@
-<?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
+<?xml version="1.0" encoding="utf-8"?><!--<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"--><!--xmlns:app="http://schemas.android.com/apk/res-auto"--><!--xmlns:tools="http://schemas.android.com/tools"--><!--android:layout_width="match_parent"--><!--android:layout_height="match_parent"--><!--tools:context=".GameList">-->
```
Comment (roosemberth) [inline app/src/main/res/layout/activity_game_list.xml]: Remove unnecessary comment

File: app/src/main/res/layout/activity_game_list.xml [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
-</android.support.constraint.ConstraintLayout>
\ No newline at end of file
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recyclerView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:scrollbars="vertical">
+
+    </android.support.v7.widget.RecyclerView>
+</LinearLayout>
```
Comment (roosemberth) [inline app/src/main/res/layout/activity_game_list.xml]: Misssing final newline, please add it given that you're modifying this file

File: app/src/main/res/layout/item_row.xml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                    app:layout_constraintRight_toRightOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    tools:text="Warhammer" />
+            </android.support.constraint.ConstraintLayout>
+        </LinearLayout>
+    </LinearLayout>
+
+</android.support.v7.widget.CardView>
+
+    <!--android:textColor="@color/textColor"-->
```
Comment (roosemberth) [inline app/src/main/res/layout/item_row.xml]: Please remove this line

File: app/build.gradle [hunk 5]
```diff
@@ -28,12 +28,18 @@ android {
...
+        targetCompatibility = '1.8'
+    }
 }
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'com.android.support:appcompat-v7:28.0.0'
     implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    implementation 'com.android.support:recyclerview-v7:28.0.0'
+    implementation 'com.android.support:cardview-v7:28.0.0'
```
Comment (roosemberth) [inline app/build.gradle]: Please sort this

File: app/src/androidTest/java/ch/epfl/sweng/erpa/GameListTest.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+        assertTrue(itemCount >= 5);
+    }
+
+    @Test
+    public void testMaxNumberOfCardsDisplayed() {
+        RecyclerView view = intentsTestRule.getActivity().findViewById(R.id.recyclerView);
+        int itemCount = view.getLayoutManager().getItemCount();
+
+        assertTrue(itemCount <= 25);
```
Comment (roosemberth) [inline app/src/androidTest/java/ch/epfl/sweng/erpa/GameListTest.java]: Where is this magic value coming from?

File: app/src/androidTest/java/ch/epfl/sweng/erpa/GameListTest.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            }
+        }
+        return ret;
+    }
+
+    @Test
+    public void testScrolling() {
+        RecyclerView view = intentsTestRule.getActivity().findViewById(R.id.recyclerView);
+        assertTrue(view.getLayoutManager().canScrollVertically());
+        assertTrue(!view.getLayoutManager().canScrollHorizontally());
```
Comment (roosemberth) [inline app/src/androidTest/java/ch/epfl/sweng/erpa/GameListTest.java]: use `assertFalse`

File: app/src/main/res/layout/activity_game_list.xml [hunk 8]
```diff
@@ -1,9 +1,19 @@
-<?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
+<?xml version="1.0" encoding="utf-8"?><!--<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"--><!--xmlns:app="http://schemas.android.com/apk/res-auto"--><!--xmlns:tools="http://schemas.android.com/tools"--><!--android:layout_width="match_parent"--><!--android:layout_height="match_parent"--><!--tools:context=".GameList">-->
+
+<!--</android.support.constraint.ConstraintLayout>-->
```
Comment (roosemberth) [inline app/src/main/res/layout/activity_game_list.xml]: Remove unnecessary comment

File: app/src/androidTest/java/ch/epfl/sweng/erpa/GameListTest.java (hunk 6 above)
Comment (rykeuse) [reply app/src/androidTest/java/ch/epfl/sweng/erpa/GameListTest.java]: It's a magic value I picked for testing as the real thing isn't implemented yet (still waiting on #41 )

Comment (roosemberth) [reply app/src/androidTest/java/ch/epfl/sweng/erpa/GameListTest.java]: Can you please add a comment on what this number means and how to replace once #41 is implemented?

File: app/src/androidTest/java/ch/epfl/sweng/erpa/activities/GameListActivityTest.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    @Test
+    public void testMinNumberOfCardsDisplayed() {
+        RecyclerView view = intentsTestRule.getActivity().findViewById(R.id.recyclerView);
+        int itemCount = view.getLayoutManager().getItemCount();
+
+        // magic number fits example in createListData in GameListActivity
+        assertTrue(itemCount >= 5);
+    }
+
+    @Test
```
[the remark this answers is not in the document]
Comment (roosemberth) [reply app/src/androidTest/java/ch/epfl/sweng/erpa/activities/GameListActivityTest.java]: Test behavioral statements

**Comment by roosemberth** [comment]
I accidentally merged (and the reset this branch). Please open a new PR to merge this branch.

In the feature, please rename the title of your PR to start with `WIP` to prevent accidentally merging an unfinished branch.

**Comment by roosemberth** [comment]
@rykeuse @DonnA98 ^^

**Pull request merged**